src/PhotometryModel.cc

```diff
@@ -0,0 +1,38 @@
+#include "lsst/log/Log.h"
+
+#include "lsst/jointcal/CcdImage.h"
+#include "lsst/jointcal/PhotometryModel.h"
+
+namespace lsst {
+namespace jointcal {
+
+bool PhotometryModel::validate(CcdImageList const &ccdImageList) const {
+    bool check = true;
+    for (auto const &ccdImage : ccdImageList) {
+        check &= checkPositiveOnBBox(*ccdImage);
+    }
+    return check;
+}
+
+bool PhotometryModel::checkPositiveOnBBox(CcdImage const &ccdImage) const {
+    bool check = true;
+    double result;
+    auto bbox = ccdImage.getImageFrame();
+    std::vector<double> xRange = {bbox.xMin, bbox.getCenter().x, bbox.xMax};
+    std::vector<double> yRange = {bbox.yMin, bbox.getCenter().y, bbox.yMax};
+    for (auto const &x : xRange)
+        for (auto const &y : yRange) {
+            // flux, fluxErr, instFlux, instFluxErr all 1
+            jointcal::MeasuredStar star(x, y, 1, 1, 1, 1);
+            result = transform(ccdImage, star);
+            if (result < 0) {
+                LOGLS_WARN(_log, "CcdImage " << ccdImage.getName() << " is negative at (" << x << "," << y
+                                             << "): " << result);
+                check = false;
```

Comment:
Short-circuit?

Reply:
No, I don't want this to short circuit: I want it to log every such case. I'll add a comment.